src/plugin.validation.dns.godaddy/Godaddy/DnsManagementClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using PKISharp.WACS.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PKISharp.WACS.Plugins.ValidationPlugins.Godaddy
+{
+
+    public class DnsManagementClient
+    {
+        private readonly string _apiKey;
+        private readonly ILogService _logService;
+        private readonly string uri = "https://api.godaddy.com/";
+
+        public DnsManagementClient(string apiKey, ILogService logService)
+        {
+            _apiKey = apiKey;
+            _logService = logService;
+        }
+
+        public async Task CreateRecord(string record, string identifier, RecordType type, string value)
+        {
+            using (var client = new HttpClient { BaseAddress = new Uri(uri) })
```

Comment:
Please don't construct your own `HttpClient` but get one from the `ProxyService` so that user proxy settings are respected and request logging happens in line with other parts of the code.